src/policy_engine/mod.rs

```diff
@@ -27,4 +27,6 @@ pub trait PolicyEngine {
         reference_data_map: HashMap<String, Vec<String>>,
         input: String,
     ) -> Result<(bool, String)>;
+
+    fn set_policy(&mut self, policy_base64_string: String) -> Result<()>;
```

Comment:
I suggest change this API to `async`. And use `tokio::fs::write` to impl this to avoid blocking calling. How do you like?